src/js/components/Articles.js

```diff
@@ -1,46 +1,99 @@
-export function Articles($el) {
+import {findAllByVideoIds} from '../apis/youtubeApis.js';
+import {getSavedVideos, subscribeStore} from '../store/videoStore.js';
+import router from '../router.js';
+
+/**
+ * @param $el
+ * @param props
+ * @param {string} props.routePath
+ * @constructor
+ */
+export function Articles($el, props) {
+
+    const state = {
+        articles: [],
+    };
+
+    const setState = ({articles}) => {
+        state.articles = articles ?? state.articles;
```

Comment:
```suggestion
        if (!articles) return;
        state.articles = articles;
        render();
```

articles가 없을땐 굳이 다시 state.articles를 재할당해 줄 필요가 없고 상태가 변하지 않았으니 render도 생략해주며 성능최적화까지 챙길 수 있을것 같은데 위 처럼 처리하면 어떨까요ㅎㅎ